recipe-client-addon/test/browser/browser_RecipeRunner.js

```diff
@@ -152,17 +154,45 @@ add_task(async function testStart() {
     ["extensions.shield-recipe-client.experiments.lazy_classify", false],
   ]});
   ok(!getStub.called, "getClientClassification hasn't been called");
-  await RecipeRunner.start();
-  ok(getStub.called, "getClientClassfication was called eagerly");
+  await RecipeRunner.run();
+  ok(getStub.called, "getClientClassification was called eagerly");
 
   // When the experiment pref is true, do not eagerly call getClientClassification.
   await SpecialPowers.pushPrefEnv({set: [
     ["extensions.shield-recipe-client.experiments.lazy_classify", true],
   ]});
   getStub.reset();
   ok(!getStub.called, "getClientClassification hasn't been called");
-  await RecipeRunner.start();
-  ok(!getStub.called, "getClientClassfication was not called eagerly");
+  await RecipeRunner.run();
+  ok(!getStub.called, "getClientClassification was not called eagerly");
 
   getStub.restore();
 });
+
+add_task(async function testStartup() {
+  const runStub = sinon.stub(RecipeRunner, "run");
+  const addCleanupHandlerStub = sinon.stub(CleanupManager, "addCleanupHandler");
+  const updateRunIntervalStub = sinon.stub(RecipeRunner, "updateRunInterval");
+
+  // in dev mode
+  await SpecialPowers.pushPrefEnv({set: [["extensions.shield-recipe-client.dev_mode", true]]});
+  RecipeRunner.init();
+  ok(runStub.called, "RecipeRunner.run is called immediately when in dev mode");
+  ok(addCleanupHandlerStub.called, "A cleanup function is registered when in dev mode");
+  ok(updateRunIntervalStub.called, "A timer is registered when in dev mode");
+
+  runStub.reset();
+  addCleanupHandlerStub.reset();
+  updateRunIntervalStub.reset();
+
+  // not in dev mode
+  await SpecialPowers.pushPrefEnv({set: [["extensions.shield-recipe-client.dev_mode", false]]});
+  RecipeRunner.init();
+  ok(!runStub.called, "RecipeRunner.run is not called immediately when not in dev mode");
+  ok(addCleanupHandlerStub.called, "A cleanup function is registered when not in dev mode");
+  ok(updateRunIntervalStub.called, "A timer is registered when not in dev mode");
+
+  runStub.restore();
+  addCleanupHandlerStub.restore();
+  updateRunIntervalStub.restore();
+});
```

Comment:
I tested that the timer works the way I want by fiddling the pref. I also tested that uninstalling the add-on didn't cause any weird errors, including when the timer would have fired. I haven't tested that there aren't unexpected changes to other times, but I'd expect that to be out of scope of Normandy's test cases.